src/cose/decoder.rs

```diff
@@ -135,24 +120,25 @@ pub fn decode_signature(bytes: Vec<u8>, payload: &[u8]) -> Result<Vec<CoseSignat
 
     // Read the key ID from the unprotected header.
     let unprotected_signature_header = &cose_signature[1];
-    let key_id = match unprotected_signature_header {
+    let (key_id_key, key_id_value) = match unprotected_signature_header {
         &CborType::Map(ref value) => {
             if value.len() < 1 {
                 return Err("Invalid COSE signature: unprotected header is empty.");
             }
-            value[0].clone()
+            let mut tmp = value.iter();
```

Comment:
This should be analogous to getting the signature algorithm, right? (i.e. we can probably just use the map lookup directly rather than an iter (this also is looking like a good piece of functionality to factor out))

Reply:
done